.github/build/Containerfile

```diff
@@ -5,10 +5,20 @@ LABEL summary="Toolchain for running pre-commit hooks." \
     io.k8s.display-name="Pre-Commit Toolchain"
 
 USER root
-RUN dnf install nodejs
+RUN dnf install nodejs -y && \
+    dnf clean all && \
+    rm -rf /var/cache/dnf
+ADD https://mirror.openshift.com/pub/openshift-v4/clients/oc/latest/linux/oc.tar.gz $TMPDIR/
+RUN tar -C /usr/local/bin -xvf $TMPDIR/oc.tar.gz && \
+    chmod +x /usr/local/bin/oc && \
+    rm $TMPDIR/oc.tar.gz
 USER $USERID
 
-COPY requirements-dev.txt /tmp/requirements-dev.txt
-RUN python -m pip install -r /tmp/requirements-dev.txt
+RUN pip3 install poetry
+COPY pyproject.toml ./
+RUN poetry install
 
-CMD bash
+#COPY requirements.txt /tmp/requirements.txt
+#RUN python -m pip install -r /tmp/requirements.txt
+
+CMD bash
```

Comment:
Missing newline at EOF

Reply:
Also curious, why are the steps and requirements for the pre-commit image changing so much? It seems like before all we needed were the contents of `requirements-dev.txt`:
```
pre-commit
poetry
pytest
pytest-mock
coverage
black==22.3.0
```
but now we are doing a lot more than just those

Reply:
Thanks for the review @Maxusmusti 
In an attempt to test out using poetry instead of pip I copied the repos pyproject.toml file in here. If we want to go that way I can reduce it to just what is needed. 
The requirements-dev file was removed and its contents added to the requirements.txt, possibly to reduce the number of files - @anishasthana might be able to confirm this. The requirements file will be removed if we go with using the toml file instead.

Reply:
@anishasthana thoughts? This is specifically for requirements for the pre-commit image. Also @Fiona-Waters would it be possible to pull these requirements in any way from the root dir pyproject, since they should exactly align with the test dependencies we have, that way we don't have to manage them in multiple places?

Reply:
I tried but wasn't able to pull them and use them through the Containerfile. Open to suggestions!

Reply:
In my opinion if we can have just one file to deal with, that'd be simplest.

Agreed on pulling from the root directory if possible. What if we ran the image build from the root directory? Would it give us access to all the stuff?

So like `podman build -f .github/build/Containerfile .` from the root path. I think it'll give you access to the complete context, so no copy needed.

Reply:
Thanks @anishasthana I've run this and it built successfully copying the pyproject.toml and resolving dependencies. Where should this change of building from root path be documented, other than in the readme?
We will also need to update the [coverage-badge.yaml](https://github.com/project-codeflare/codeflare-sdk/blob/4593dd5cd07c0a737150961302ae5ed76ed6da68/.github/workflows/coverage-badge.yaml#L26) to account for the removal of the requirements.txt file